src/sendquery.js

```diff
@@ -51,9 +51,9 @@ async function execute(email, key, project, query, service, params = {
         return true;
       };
```

Comment:
Hmm,  not sure I can follow. The max size should be a defined constant. 

And why not keep track of the current result size, increment with the actual row size and check threshold on every row?

Reply:
The size operation is too slow; running it for every row is expensive. Approximating using ten rows; then just multiplying that estimate by the current length is a good approximation!